### PR TITLE
Fix desync

### PR DESCRIPTION
**Pull request by Raycoms**
# Changes proposed in this pull request:
- Fixes client/server desync (also improves server side performance, by writing colony only once to NBT)
- Dman ninja pickup fix (dman has to walk until the building to pickup now)
- Dman gathering crafter output prematurely (only on specific pickup)
- Fix stonemason recipe (allow stonebricks, stairs, etc, don't allow redstone, etc)

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/colony/requestsystem/management/manager/StandardRequestManager.java [hunk 1]
```diff
@@ -175,12 +175,13 @@ private void markDirty()
     @Override
     public boolean isDirty()
     {
-        if (dirty)
```
Comment (marchermans) [inline src/main/java/com/minecolonies/coremod/colony/requestsystem/management/manager/StandardRequestManager.java:178]: This is a half assed hotfix(?!??) for something without description. And I am 100% sure this breaks a lot of things. Please explain why this change is needed and how it prevents your not described actual concrete problem you are trying to fix!

**Review by marchermans — CHANGES_REQUESTED** [review]
More explanation needed

**Comment by Raycoms** [comment]
Atm we reset dirty when checking if dirty, which can break if we check
dirty two times (which is the actual problem with the client server sync)
Besides that we serialize the colony one time each person, which is also a
lot of lost performance

**Review by marchermans — APPROVED** [review]
